### PR TITLE
Tree mesh accel patch

### PR DESCRIPTION
**Pull request by fourndo**
**Various fixes for the new implementation of TreeMesh**
- Add utility to build mesh around point, surface
- Add example with topography
- Update PF
- Fix bugs in auto-generation of examples

Various fixes to docstring related bugs were merged from PR #684 . The highlights of this branch are:

* [x] forward modeling and inversion example from VRM code with figures
* [x] add properties to VRM module using properties package
* [x] generate API for VRM
* [x] add square loop receiver class for VRM

### CODE REVIEW

**Comment by dccowan** [comment]
image

Not sure why this is done.

**Comment by dccowan** [comment]
image

Codacy-related error. Not sure what this does

**Comment by dccowan** [comment]
image

If Solver is not imported again as SimpegSolver, the test 'test_mag_inversion_linear.py' will break. Not sure why this is

File: examples/00_published/plot_booky_1D_time_freq_inv.py [hunk 1]
```diff
@@ -457,6 +457,8 @@ def run(plotIt=True, saveFig=False, cleanup=True):
     if plotIt:
         plt.show()
 
+    resolve.close()
+    skytem.close()
```
Comment (fourndo) [inline examples/00_published/plot_booky_1D_time_freq_inv.py]: Nice catch

**Comment by jcapriot** [comment]
So this is just a thought I had, but feel free to ponder on it.

I feel like most of this PR should actually be a separate PR onto discretize. I.e. most of the mesh building functions.

**Comment by fourndo** [comment]
You are probably right yea. I can move the creation utils on different PR and deal with the plotSections with vectors there as well. Let's talk on Tuesday.

**Comment by lheagy** [comment]
@dccowan: can you please also include a description of the changes that were included in the VRM docstrings updates? These will be included in the release

**Comment by lheagy** [comment]
@fourndo: there are two examples that I am getting errors on when I run them

- examples/02-mesh/plot_quadtree_facediv.py
```
 File "...../examples/02-mesh/plot_quadtree_facediv.py", line 14, in run
    M._refineCell([0, 0, 0])
AttributeError: 'TreeMesh' object has no attribute '_refineCell'
```

- examples/04-grav/plot_inversion_linear.py
```
File "...../examples/04-grav/plot_inversion_linear.py", line 154, in run
    PF.Gravity.plot_obs_2D(rxLoc, d=data)
AttributeError: module 'SimPEG.PF.Gravity' has no attribute 'plot_obs_2D'
```
Would you mind taking a look?

**Comment by fourndo** [comment]
Changes will be split into two different PR

**Pull request closed without merging**